epochStart/metachain/rewards.go

```diff
@@ -223,19 +237,42 @@ func (rc *rewardsCreator) addValidatorRewardsToMiniBlocks(
 		}
 
 		rc.accumulatedRewards.Add(rc.accumulatedRewards, rwdTx.Value)
-		shardId := rc.shardCoordinator.ComputeId([]byte(rwdInfo.address))
-		if shardId == core.MetachainShardId {
-			protocolSustainabilityRwdTx.Value.Add(protocolSustainabilityRwdTx.Value, rwdTx.Value)
-			continue
+		mbId := rc.shardCoordinator.ComputeId([]byte(rwdInfo.address))
+		if mbId == core.MetachainShardId {
+			mbId = rc.shardCoordinator.NumberOfShards()
+
+			if !rc.flagDelegationSystemSCEnabled.IsSet() || !rc.isSystemDelegationSC(rwdTx.RcvAddr) {
+				protocolSustainabilityRwdTx.Value.Add(protocolSustainabilityRwdTx.Value, rwdTx.GetValue())
+				continue
+			}
 		}
 
 		rc.currTxs.AddTx(rwdTxHash, rwdTx)
-		miniBlocks[shardId].TxHashes = append(miniBlocks[shardId].TxHashes, rwdTxHash)
+		miniBlocks[mbId].TxHashes = append(miniBlocks[mbId].TxHashes, rwdTxHash)
 	}
 
 	return nil
 }
 
+func (rc *rewardsCreator) isSystemDelegationSC(address []byte) bool {
+	acc, errExist := rc.userAccountsDB.GetExistingAccount(address)
+	if errExist != nil {
+		return false
+	}
+
+	userAcc, ok := acc.(state.UserAccountHandler)
+	if !ok {
+		return false
+	}
+
+	val, err := userAcc.DataTrieTracker().RetrieveValue([]byte(core.DelegationSystemSCKey))
```

Comment:
because the given address can be the staking smart contract, the esdt contract or other system scs. Only delegtion system SC addresses are needed here.

Reply:
No. This cannot be faked. Noone can create accounts in the metachain.